src/index.ts

```diff
@@ -1,17 +1,206 @@
 import {
-  JupyterFrontEnd,
-  JupyterFrontEndPlugin
+    ILayoutRestorer,
+    JupyterFrontEnd,
+    JupyterFrontEndPlugin
 } from '@jupyterlab/application';
 
-/**
- * Initialization data for the image-editor extension.
- */
-const plugin: JupyterFrontEndPlugin<void> = {
-  id: 'image-editor:plugin',
-  autoStart: true,
-  activate: (app: JupyterFrontEnd) => {
-    console.log('JupyterLab extension image-editor is activated!');
-  }
+import {
+    CommandToolbarButton,
+    ICommandPalette,
+    WidgetTracker,
+    IWidgetTracker
+} from '@jupyterlab/apputils';
+
+import { Token } from '@lumino/coreutils';
+import { refreshIcon } from  '@jupyterlab/ui-components';
+
+import { ImageEditorDocumentWidget } from './editor';  
+import { ImageEditorFactory } from './factory';
+
+const FACTORY = 'ImageEditor';
+
+namespace CommandIDs {
+  export const rotateClockwise = 'image-editor:rotate-clockwise';
+  export const crop = 'image-editor:crop';
+  export const applyCrop = 'image-editor:apply-crop';
+  export const cancelCrop = 'image-editor:cancel-crop';
+}
+
+type IImageEditorTracker = IWidgetTracker<ImageEditorDocumentWidget>;
+export const IImageEditorTracker = new Token<IImageEditorTracker>('image-editor/tracki');
+
+const extension: JupyterFrontEndPlugin<IImageEditorTracker> = {
+    id: '@jupyterlab/image-editor-extension:plugin',
+    autoStart: true,
+    requires: [ILayoutRestorer, ICommandPalette],
```

Comment:
it is better to put the palette and the restorer as optional dependencies. This will allow the extension to be used in retrolab for example.

```suggestion
    optional: [ILayoutRestorer, ICommandPalette],
```